fastlane/github-actions
# Add `communicate-on-pull-request-released` GitHub Action

### Description

The action will:

* add a comment to the pull request about including the changes in a release
```
Congratulations! 🎉 This was released as part of fastlane x.y.z 🚀
```
* add `status: released` label to the pull request and remove `status: included-in-next-release` label from it if existing
* add a comment to the referenced issue that the pull request has been released
```
The pull request #xxxxx that closed this issue was merged and released as part of fastlane x.y.z 🚀
Please let us know if the functionality works as expected as a reply here. If it does not, please open a new issue. Thanks!
```

Please note that this is just an action implementation. It does not take any effect itself. A new workflow which uses the action will be created.

### Testing

**Manual testing**

A workflow file used for a manual testing:

```yml
name: Processing release
on: 
  release:
    types: [published]

jobs:   
  communicate-on-pull-request-released:
    name: PR released
    runs-on: ubuntu-latest    
    steps:
    - uses: actions/checkout@master
      with:
        ref: refs/heads/master
    - uses: mollyIV/github-actions/communicate-on-pull-request-released@communicate-on-pull-request-released
      with:
        repo-token: "${{ secrets.BOT_TOKEN }}"
```

Please not that the action trigger is a [`release [published]`](https://help.github.com/en/github/automating-your-workflow-with-github-actions/events-that-trigger-workflows#release-event-release).

My bot is set to my private account, that's why you see the comments from the author.

![image](https://user-images.githubusercontent.com/10795657/67466378-46a77000-f647-11e9-8efb-d8a60f6c342d.png)

A comment added to pull requests number 4 and 6. 

![image](https://user-images.githubusercontent.com/10795657/67466467-6a6ab600-f647-11e9-870e-aee6cc953245.png)

A comment added to a referenced issue number 1 from a pull request description's.

![image](https://user-images.githubusercontent.com/10795657/67466532-82dad080-f647-11e9-996c-68994e39f587.png)

🎉 

**Unit testing**

To test the changes, run unit tests by executing npm test (followed by npm install to install the package dependencies).
